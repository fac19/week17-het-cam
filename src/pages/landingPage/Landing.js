import React from "react";
import { Link } from "react-router-dom";
import Heading from "../../components/global/heading/Heading";
import { useHistory } from "react-router-dom";
import { TextButton } from "../../components/global/buttons/Buttons";
import netlifyIdentity from "netlify-identity-widget";
import { makeStyles } from "@material-ui/core/styles";
import { LandingContainer, LandingWrapper, StyledImg } from "./Landing.style";
import ImgWrapper from "../../components/global/images/ImgWrapper";
import logo from "./logo.png";
import rocket from "./rocket.svg";
import { accentColour, linkColour } from "../../components/global/colours";

const useStyles = makeStyles({
  links: {
    fontFamily: "Josefin Sans, sans-serif",
    transition: "all 0.2s ease-in-out",
    color: linkColour,
    textDecoration: "underline",
    fontSize: "1.5rem",
    "&:hover": {
      fontSize: "1.8rem",
      color: accentColour,
    },
  },
});

const Landing = () => {
  const classes = useStyles();

  const history = useHistory();
  return (
    <LandingContainer>
      <Heading>Enterprise Adventure</Heading>
      <LandingWrapper>
        <StyledImg src={rocket} alt="rocket" />
        <Link to="/about" className={classes.links}>
          Learn More About the App
        </Link>
        <TextButton
          text={"Join the Movement!"}
          onClick={() => {
            netlifyIdentity.open();
            history.push(`/signup`);
          }}
        />
      </LandingWrapper>
      <ImgWrapper>
        <img src={logo} alt={"Teach a Man to Fish logo"} />
      </ImgWrapper>
    </LandingContainer>
  );
};

export default Landing;
