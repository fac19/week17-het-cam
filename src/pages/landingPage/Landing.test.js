import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Landing from "./Landing";

test("Landing page component is styled", () => {
  const tree = renderer
    .create(
      <Router>
        <Landing />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
  expect(tree).toHaveStyleRule("height", "80vh");
  expect(tree).toHaveStyleRule("justify-content", "space-between");
});

test("Join the movement button renders correctly", () => {
  render(
    <Router>
      <Landing />
    </Router>,
  );
  screen.getByRole("button");
  screen.getByText("Join the Movement!");
});

test("Learn More About the App link renders correctly", () => {
  render(
    <Router>
      <Landing />
    </Router>,
  );
  screen.getByRole("link");
  screen.getByText("Learn More About the App");
});

test("TAMTF logo renders correctly", () => {
  render(
    <Router>
      <Landing />
    </Router>,
  );
  screen.getByAltText("Teach a Man to Fish logo");
});
