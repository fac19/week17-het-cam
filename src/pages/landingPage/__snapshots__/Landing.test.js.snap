// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component is styled 1`] = `
.c1 {
  font-family: "Josefin Sans",sans-serif;
  font-size: 2rem;
  color: #032049;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 80vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  height: 100%;
  width: 100%;
}

.c3 {
  padding: 1rem;
  width: 6rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  text-align: center;
}

@media (min-width:600px) {
  .c1 {
    font-size: 4rem;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 8rem;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Enterprise Adventure
  </h1>
  <section
    className="c2"
  >
    <img
      alt="rocket"
      className="c3"
      src="rocket.svg"
    />
    <a
      className="makeStyles-links-1"
      href="/about"
      onClick={[Function]}
    >
      Learn More About the App
    </a>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-2"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Join the Movement!
      </span>
    </button>
  </section>
  <div
    className="c4"
  >
    <img
      alt="Teach a Man to Fish logo"
      src="logo.png"
    />
  </div>
</section>
`;
